src/Files.App/UserControls/NavigationToolbar.xaml

```diff
@@ -50,585 +49,426 @@
+							IsTabStop="True" />
+					</Flyout>
+				</Button.Flyout>
+
+			</Button>
+
+			<!--  Update The Files App Button When An Update Available  -->
+			<Button
+				x:Name="UpdateAppButton"
+				HorizontalContentAlignment="Stretch"
+				VerticalContentAlignment="Stretch"
+				x:Load="{x:Bind ViewModel.IsUpdateAvailable, Mode=OneWay}"
+				AccessKey="1"
+				AccessKeyInvoked="Button_AccessKeyInvoked"
+				AutomationProperties.Name="{helpers:ResourceString Name=UpdateFiles}"
+				Command="{x:Bind ViewModel.UpdateCommand, Mode=OneWay}"
+				IsEnabled="{x:Bind ViewModel.IsUpdating, Mode=OneWay, Converter={StaticResource BoolNegationConverter}}"
+				Style="{StaticResource AddressToolbarButtonStyle}"
+				ToolTipService.ToolTip="{helpers:ResourceString Name=UpdateFiles}">
+				<Grid>
+					<!--  Icon  -->
 					<controls:ThemedIcon
+						x:Name="UpdateIcon"
 						Width="16"
 						Height="16"
-						Style="{StaticResource App.ThemedIcons.AppUpdatedBox}" />
-				</Button>
-			</StackPanel>
-
-			<!--  Status Center TeachingTip  -->
-			<TeachingTip
-				x:Name="StatusCenterTeachingTip"
-				Title="{helpers:ResourceString Name=OngoingTasksTeachingTip/Title}"
-				IsOpen="False"
-				Subtitle="{helpers:ResourceString Name=OngoingTasksTeachingTip/Subtitle}"
-				Target="{x:Bind ShowStatusCenterButton}" />
-		</Grid>
+						x:Load="{x:Bind ViewModel.IsUpdating, Mode=OneWay, Converter={StaticResource BoolNegationConverter}}"
+						IsFilled="True"
+						Style="{StaticResource App.ThemedIcons.AppUpdateAvailable}" />
+
+					<!--  Progress  -->
+					<ProgressRing
+						x:Name="UpdateProgressRing"
+						Width="20"
+						Height="20"
+						x:Load="{x:Bind ViewModel.IsUpdating, Mode=OneWay}"
+						IsIndeterminate="True" />
+				</Grid>
+			</Button>
+
+			<!--  Open the Release Notes dialog  -->
+			<Button
+				x:Name="OpenReleaseNotesButton"
+				HorizontalContentAlignment="Stretch"
+				VerticalContentAlignment="Stretch"
+				x:Load="{x:Bind ViewModel.IsAppUpdated, Mode=OneWay}"
+				AccessKey="2"
+				AccessKeyInvoked="Button_AccessKeyInvoked"
+				AutomationProperties.Name="{x:Bind Commands.OpenReleaseNotes.Label, Mode=OneWay}"
+				Command="{x:Bind Commands.OpenReleaseNotes, Mode=OneWay}"
+				Style="{StaticResource AddressToolbarButtonStyle}"
+				ToolTipService.ToolTip="{x:Bind Commands.OpenReleaseNotes.LabelWithHotKey, Mode=OneWay}"
+				Visibility="{x:Bind Commands.OpenReleaseNotes.IsExecutable, Mode=OneWay}">
+				<controls:ThemedIcon
+					Width="16"
+					Height="16"
+					Style="{StaticResource App.ThemedIcons.AppUpdatedBox}" />
+			</Button>
+		</StackPanel>
+
+		<!--  Status Center TeachingTip  -->
+		<TeachingTip
+			x:Name="StatusCenterTeachingTip"
+			Title="{helpers:ResourceString Name=OngoingTasksTeachingTip/Title}"
+			IsOpen="False"
+			Subtitle="{helpers:ResourceString Name=OngoingTasksTeachingTip/Subtitle}"
+			Target="{x:Bind ShowStatusCenterButton}" />
 
 		<VisualStateManager.VisualStateGroups>
```

Comment:
Can you confirm if the visual states for hiding the search bar are still working properly?

Reply:
Yeah it works but due to #16930, the search glyph was not shown. I already fixed in the latest commit though.

![image](https://github.com/user-attachments/assets/4fee4bb2-cf41-4311-b637-80ef6d42da8e)

In any event, we'll no longer need this prop, there shouldnt be a problem here to fix.

Reply:
Thanks.